### PR TITLE
[WIP] feat(cmo): add monitoring to CMO

### PR DESCRIPTION
**Pull request by zdtsw**
- skip blackbox part, no need in CMO
- make segment-io only for Managed offering
- move old monitoring networkpolicy into manifests, the way to deploy them are the same logic, no changed
- component PromRule is managed by operator, live in component folder
- CMO config only create or remove by monitoring.managementstate, live in dsci folder
- move monitoring function into obo package
- limitation from alertmanagerconfig which does not support built-in gotemp, have to make email as configmap then runtime inject into CR

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: config/networkpolicy/monitoring/monitoring.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: redhat-ods-monitoring
+  namespace: redhat-ods-monitoring
+spec:
+  podSelector: {}
+  ingress:
+    - ports:
+        - protocol: TCP
+          port: 443
```
Comment (aslakknutsen) [inline config/networkpolicy/monitoring/monitoring.yaml:18]: Is this all the ports that some component possible could have something to monitor on? A set of standardized ports could have some value, or possible open it up to be defined by the component it self.

Comment (zdtsw) [reply config/networkpolicy/monitoring/monitoring.yaml:18]: these are the ones copied over from what we are using now, not refined anything in the draft yet.
but i think the goal is to be able to migrate function first. we can do the change independently,

**Comment by AjayJagan** [comment]
Hi @zdtsw , in the description you have mentioned that blackbox exporter is not needed. But then how do we scrape the metrics of [these endpoints.](https://github.com/red-hat-data-services/rhods-operator/blob/main/config/monitoring/prometheus/apps/prometheus-configs.yaml#L44-L130). 
IIUC, it is checking the health of those endpoints through the bb exporter.
Let me know if I am wrong here

**Comment by zdtsw** [comment]
no need continue review on this draft, code has been cherry-picked into https://github.com/opendatahub-io/opendatahub-operator/pull/1025 for further work

File: controllers/dscinitialization/observibility_setup.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+func (r *DSCInitializationReconciler) observabilityCapabilityFeatures(instance *dsciv1.DSCInitialization) feature.FeaturesProvider {
+	return func(handler *feature.FeaturesHandler) error {
+		monitoringNamespace := instance.Spec.Monitoring.Namespace
+		alertmanageErr := feature.CreateFeature("create-alertmanager").
+			For(handler).
+			ManifestSource(Templates.Source).
+			Manifests(
+				path.Join(Templates.AlertManageDir),
+			).
+			WithData(obo.AlertmanagerDataValue). // fill in alertmanager data
```
Comment (AjayJagan) [inline controllers/dscinitialization/observibility_setup.go:96]: one issue I found out here(for the config map) is that, the `WithData()` where we inject the template from the configmap is called before the manifests are applied. So I we wont have the configmap ready when withData is called. Alternative solution could be to apply the configmap in precondition?

Comment (zdtsw) [reply controllers/dscinitialization/observibility_setup.go:96]: i saw this part yesterday but not done anything to fix it yet. 
i am still thinking

Comment (zdtsw) [reply controllers/dscinitialization/observibility_setup.go]: this https://github.com/opendatahub-io/opendatahub-operator/pull/1004/commits/3c39f4fcff076e0b4697c7b5574ae21e505420c1 should have it fixed as an ugly workaround.

Comment (AjayJagan) [reply controllers/dscinitialization/observibility_setup.go]: cool!

**Pull request closed without merging**